mmv1/products/netapp/volume.yaml

```diff
@@ -0,0 +1,405 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Volume'
+description: |
+  A volume is a file system container in a storage pool that stores application, database, and user data.
+
+  You can create a volume's capacity using the available capacity in the storage pool and you can define and resize the capacity without disruption to any processes.
+
+  Storage pool settings apply to the volumes contained within them automatically.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Quickstart': 'https://cloud.google.com/netapp/volumes/docs/get-started/quickstarts/create-volume'
+    'Documentation': 'https://cloud.google.com/netapp/volumes/docs/configure-and-use/volumes/overview'
+  api: 'https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.volumes'
+base_url: projects/{{project}}/locations/{{location}}/volumes
+self_link: projects/{{project}}/locations/{{location}}/volumes/{{name}}
+create_url: projects/{{project}}/locations/{{location}}/volumes?volumeId={{name}}
+update_url: projects/{{project}}/locations/{{location}}/volumes/{{name}}
+update_verb: :PATCH
+update_mask: true
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+id_format: 'projects/{{project}}/locations/{{location}}/volumes/{{name}}'
+import_format: ['projects/{{project}}/locations/{{location}}/volumes/{{name}}']
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Name of the pool location. Usually a region name, expect for some STANDARD service level pools which require a zone name.
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description:
+      The name of the volume. Needs to be unique per location.
+    required: true
+    immutable: true
+    url_param_only: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'volume_basic'
+    primary_resource_id: 'test_volume'
+    vars:
+      volume_name: 'test-volume'
+      pool_name: 'test-pool'
+      network_name: 'test-network'
+      global_name: 'test-address'
+    test_vars_overrides:
+      network_name: 'acctest.BootstrapSharedServiceNetworkingConnection(t, "gcnv-network-config-1", acctest.ServiceNetworkWithParentService("netapp.servicenetworking.goog"))'
+
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'shareName'
+    description: |
+      Share name (SMB) or export path (NFS) of the volume. Needs to be unique per location.
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: 'psaRange'
+    description: |
+      Name of the Private Service Access allocated range. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'storagePool'
```

Comment:
this seems like it would be immutable? Can we test changing it if not?

Reply:
Volumes can be assigned to a different storagePool in certain situations.
The condition set is complex and cannot be handled by the provider. Let the API return good or bad. 
https://cloud.google.com/netapp/volumes/docs/configure-and-use/volumes/edit-volume
Summary: storagePool should be mutable. Test case: create two pools in same region with same parameters (VPC, AD, CMEK, LDAP), but differing service levels (premium and extreme). Reassign a volume from one to the other.